### PR TITLE
refactor(vega-typings): avoid object-based union types for scale

### PR DESCRIPTION
**Pull request by kanitw**
Merge https://github.com/vega/vega/pull/2387 first

### CODE REVIEW

File: packages/vega-typings/types/spec/scale.d.ts [hunk 1]
```diff
@@ -78,21 +79,23 @@ export type ScaleInterpolate =
...
-      fields: ScaleField[];
-    }
-  | {
-      fields: ((string | number | boolean)[] | DataRef | SignalRef)[];
-    };
-export type ScaleData =
-  | (DataRef & { sort?: SortField })
-  | (MultiDataRef & { sort?: UnionSortField });
+
+export type ScaleMultiDataRef = {
```
Comment (domoritz) [inline packages/vega-typings/types/spec/scale.d.ts]: Why not multi scale data ref?

Comment (kanitw) [reply packages/vega-typings/types/spec/scale.d.ts]: It's single scale, not multi scale?  I'm also just treating scale as prefix here.

**Comment by domoritz** [comment]
I suppose you made this change so you can import the individual types in Vega-Lite, right?

**Comment by kanitw** [comment]
> I suppose you made this change so you can import the individual types in Vega-Lite, right?

Yes, and even if we import the main one, we want each of them to be named.  Otherwise, Altair won't generate a class and we can't refer to any of these objects in the docs.

File: packages/vega-typings/types/spec/scale.d.ts [hunk 2]
```diff
@@ -55,44 +56,64 @@ export type UnionSortField =
...
-    }
-  | {
-      fields: ((string | number | boolean)[] | DataRef | SignalRef)[];
-    };
-export type ScaleData =
-  | (DataRef & { sort?: SortField })
-  | (MultiDataRef & { sort?: UnionSortField });
+
+export interface ScaleMultiDataRef {
+  fields: ((string | number | boolean)[] | ScaleDataRef | SignalRef)[];
```
Comment (domoritz) [inline packages/vega-typings/types/spec/scale.d.ts]: Do we have a type for `string | number | boolean`?

Comment (kanitw) [reply packages/vega-typings/types/spec/scale.d.ts]: Not that I know of.

Comment (kanitw) [reply packages/vega-typings/types/spec/scale.d.ts]: Let's merge first, if you find one we can fix them later.

**Pull request merged**